recipes/gpm-imerg/recipe.py

```diff
@@ -0,0 +1,35 @@
+from pangeo_forge_recipes.recipes import XarrayZarrRecipe
+from pangeo_forge_recipes.patterns import pattern_from_file_sequence
+import pandas as pd
+
+#TODO: replace with ENV vars
+username = password = 'pangeo@developmentseed.org'
+
+dates = pd.date_range("2000-06-01T00:00:00", "2021-05-31T23:59:59", freq="30min")
+input_url_pattern = (
+    "https://arthurhouhttps.pps.eosdis.nasa.gov/gpmdata/{yyyy}/{mm}/{dd}/imerg/3B-HHR.MS.MRG.3IMERG.{yyyymmdd}-S{sh}{sm}00-E{eh}{em}59.{MMMM}.V06B.HDF5"
+)
+
+input_urls = [
+    input_url_pattern.format(
+        yyyy=hhr.strftime("%Y"),
+        mm = hhr.strftime("%m"),
+        dd = hhr.strftime("%d"),
+        yyyymmdd=hhr.strftime("%Y%m%d"),
+        sh = hhr.strftime("%H"),
+        sm = hhr.strftime("%M"),
+        eh = hhr.strftime("%H"),
+        em = (hhr+pd.Timedelta("29 min")).strftime("%M"),
+        MMMM = f'{(hhr.hour*60 + hhr.minute):04}'
+    )
+    for hhr in dates
+]
+
+pattern = pattern_from_file_sequence(input_urls, "time", nitems_per_file=1)
```

Comment:
This is a _very long_ sequence (368160 items).  Using `pattern_from_file_sequence` is probably not recommended because it does not serialize very well. So probably better to try to define the FilePattern directly, as in [this example](https://pangeo-forge.readthedocs.io/en/latest/recipe_user_guide/file_patterns.html#create-a-filepattern). (Also the ["alternative method"](https://pangeo-forge.readthedocs.io/en/latest/tutorials/xarray_zarr/netcdf_zarr_sequential.html#step-2-define-file-pattern) in the tutorial.)

This is an obscure point, and I would not have expected you to know this. You were clearly just following the tutorial as currently written.

Reply:
Meta comment: ☝️ prime example of the type of feedback I'd like to eventually be able offer automatically, as envisioned in https://github.com/pangeo-forge/pangeo-forge-recipes/issues/162#issuecomment-915581995

Reply:
Thanks for the review. I've swapped for the FilePattern method, please re-review.